include/arch/arm/arch/64/mode/kernel/thread.h

```diff
@@ -15,13 +15,25 @@ static inline word_t CONST sanitiseRegister(register_t reg, word_t v, bool_t arc
 {
     if (reg == SPSR_EL1) {
         if (archInfo) {
-            switch (v & 0x1f) {
-            case PMODE_EL0t:
-            case PMODE_EL1t:
-            case PMODE_EL1h:
-                return v;
-            default:
-                break;
+            if (v & BIT(4)) {
```

Comment:
This commit probably needs to be reordered until after we support 32-bit guests on aarch64

Reply:
removed this commit for now, I'll put it up later with other relevant changes